src/os/bluestore/BlueStore.h

```diff
@@ -153,9 +153,9 @@ class BlueStore : public ObjectStore,
     }
 
     BufferSpace *space;
-    uint16_t state;             ///< STATE_*
-    uint16_t cache_private = 0; ///< opaque (to us) value used by Cache impl
-    uint32_t flags;             ///< FLAG_*
+    uint8_t state;             ///< STATE_*
```

Comment:
This fix has no almost no sense since compiler aligns the struct members at 64 bit boundary (for 64-bit platforms) by default.  Consider following code snippet:

` #include <stdio.h>`
`#include <stdint.h>`
`struct A{`
`  uint16_t a,b;`
`  uint32_t f;`
`  uint64_t c;`
`};`
`struct B{`
`  uint8_t a,b;`
`  uint8_t f;`
`  uint64_t c;`
`};`
`struct C{`
`  uint64_t c;`
`  uint8_t a,b;`
`  uint8_t f;`
`};`
`struct __attribute__((packed, aligned(1))) BPacked{`
`  uint8_t a,b;`
`  uint8_t f;`
`  uint64_t c;`
`};`
`int main()`
`{`
`  printf("%lu %lu %lu %lu\n", sizeof(A), sizeof(B), sizeof(C), sizeof(BPacked));`
`  return 0;`
`}`

Packed structure helps a bit but still there is heap allocation granularity that is most probably brings 64-bit alignment back.